File: workspace-2025-1c/IntroPOO/src/circulo/Circulo.java
package circulo;

import io.Punto;

public class Circulo {

	private double radio;
	private Punto centro;
	private Color color;

	public Circulo(double radio, double xc, double yc, Color color) {
		this.radio = radio;
		this.centro = new Punto(xc, yc);
		this.color = color;
	}

	public Circulo(double radio, Punto centro, Color color) {
		this.radio = radio;
		this.centro = centro;
		this.color = color;
	}
	
	public Circulo() {
		this(1, new Punto(0,0), Color.BLANCO);
	}

	@Override
	public String toString() {
		return "Circulo [radio=" + radio + ", centro=" + centro +", color=" + color + "]";
	}

	public double area() {
		return Math.PI * Math.pow(radio, 2);
	}

	public double perimetro() {
		return Math.PI * this.diametro();
	}

	public double diametro() {
		return 2 * radio;
	}
	
	public double getRadio() {
		return this.radio;
	}

	public static void main(String[] args) {
		Circulo c1 = new Circulo(4, 1, 1, Color.ROJO);
		System.out.println(c1);
		System.out.println(c1.area());
		System.out.println(c1.perimetro());
		System.out.println(c1.diametro());
	}

}

File: 03-encapsulamiento/geometria/circulo.py
import math

from color import Color
from punto import Punto


class Circulo:
    def __init__(self, radio, xc, yc, color: Color):
        self.__radio = radio
        self.__centro = Punto(xc, yc)
        self.__color = color

    @property
    def radio(self):
        return self.__radio

    @radio.setter
    def radio(self, radio):
        if radio <= 0:
            raise ValueError("El radio debe ser >= 0")
        self.__radio = radio

    @property
    def centro(self):
        return self.__centro

    @property
    def color(self):
        return self.__color

    def area(self):
        return math.pi * pow(self.radio, 2)

    def perimetro(self):
        return 2 * math.pi * self.radio

    def diametro(self):
        return 2 * self.radio

    def __repr__(self):
        return f"Soy un circulo de radio {self.radio} y color {self.__color.name}"

    def desplazar(self, en_x, en_y):
        self.centro.desplazar(en_x, en_y)


def main():
    c1 = Circulo(2, 1, 1, Color.ROJO)
    c2 = Circulo(4, 2, -2, Color.AZUL)
    print(c1)
    print(c2)
    print("radio de c1: ", c1.radio)
    print("diametro de c1: ", c1.diametro())
    print(c2.perimetro())
    print(c1.color)
    print(c1.area())
    print(c2.area())


if __name__ == "__main__":
    main()
